### ISSUE TITLE
[v4] Strange behavior with input[type=checkbox]

### ISSUE THREAD

**Issue by mindeavor** [marko-js/marko#497]
## Bug Report

Server-side checkbox rendering works perfectly, but client-side has strange behavior when using two-way data binding (see code in details).

### Expected Behavior

Clicking the first checkbox should uncheck it.

### Actual Behavior

Clicking the first checkbox checks the second checkbox. Clicking a few more times results in more strange behavior (sometimes 2 clicks needed to uncheck, etc.)

### Possible Fix

A problem with the underlying virtual dom library?

<details><summary>Additional Info</summary>

### Your Environment

* Version used: 4.0.0-beta.6
* Environment name and version: Chrome 54
* Operating System and version: OS X 10.11.6

### Code to Reproduce

```html
<script>
module.exports = {
  onInput(attrs) {
    this.state = { first: true, second: false }
  },
  toggleValue(e) {
    this.state[ e.target.value ] = e.target.checked
    console.log("New state:", this.state.first, this.state.second)
  },
}
</script>

<div id="the-component">

  <input type="checkbox" name="opts[]" value="first" checked=state.first on-change('toggleValue') />
  First Checkbox
  <br />

  <input type="checkbox" name="opts[]" value="second" checked=state.second on-change('toggleValue') />
  Second Checkbox

</div>
```

</details>

**Comment by patrick-steele-idem** [comment]
Taking a look now and will update here when I know more. Thanks for reporting the problem and providing the code to reproduce.

**Comment by patrick-steele-idem** [comment]
It was indeed a bug with the virtual DOM implementation. Marko was not correctly handling different falsey values for a boolean property on a DOM element. I added a failing test using your code and fixed the problem. New version published with fix: `marko@4.0.0-beta.7`

Please let me know if you see any other issues. Thank you!
